packages/joint-core/src/dia/Paper.mjs

```diff
@@ -2029,41 +2100,25 @@ export const Paper = View.extend({
             this._updates.sort = true;
             return;
         }
-        this.sortViewsExact();
+        this.sortLayersExact();
     },
 
-    sortViewsExact: function() {
-
-        // Run insertion sort algorithm in order to efficiently sort DOM elements according to their
-        // associated model `z` attribute.
-
-        var cellNodes = Array.from(this.cells.childNodes).filter(node => node.getAttribute('model-id'));
-        var cells = this.model.get('cells');
+    sortLayersExact: function() {
+        const { _layers: { viewsMap }} = this;
 
-        sortElements(cellNodes, function(a, b) {
-            var cellA = cells.get(a.getAttribute('model-id'));
-            var cellB = cells.get(b.getAttribute('model-id'));
-            var zA = cellA.attributes.z || 0;
-            var zB = cellB.attributes.z || 0;
-            return (zA === zB) ? 0 : (zA < zB) ? -1 : 1;
-        });
+        Object.values(viewsMap).filter(view => view instanceof CellLayerView).forEach(view => view.sortExact());
     },
 
     insertView: function(view, isInitialInsert) {
-        const { el, model } = view;
+        const { model } = view;
 
-        const layerName = model.get('layer') || this.DEFAULT_CELL_LAYER;
+        const layerName = model.layer();
         const layerView = this.getLayerView(layerName);
 
-        switch (this.options.sorting) {
-            case sortingTypes.APPROX:
-                layerView.insertSortedNode(el, model.get('z'));
-                break;
-            case sortingTypes.EXACT:
-            default:
-                layerView.insertNode(el);
-                break;
-        }
+        layerView.insertCellView(view);
+
+        this.trigger('cell:inserted', view, isInitialInsert);
```

Comment:
Perhaps we should call it `'cell:insert'` (we don't use `added`, `changed` either).

Or better `'cell:mount'`. Analogous to the `onMount()` method.

Reply:
Or remove it completely, if it is not used in this PR.